src/test/java/com/emc/object/s3/S3MetadataSearchTest.java

```diff
@@ -237,7 +381,99 @@ public void testListObjectsWithEncoding() {
         }
     }
 
-    @Test // TODO: blocked by STORAGE-18866
+    @Test //TODO: blocked by STORAGE-30527
+    public void testListObjectsWithPrefixEncoding() {
+        String bucketName = getTestBucket();
+
+        String badKey = "prefix/bad\u001dkey";
+        client.putObject(new PutObjectRequest(getTestBucket(), badKey, new byte[0]).withObjectMetadata(
+                new S3ObjectMetadata().addUserMetadata("index-field", "bad-key")
+                        .addUserMetadata("field-valid", "true")
+                        .addUserMetadata("key-valid", "false")
+        ));
+
+        String goodKey = "prefix/good-key-and-field";
+        client.putObject(new PutObjectRequest(getTestBucket(), goodKey, new byte[0]).withObjectMetadata(
+                new S3ObjectMetadata().addUserMetadata("index-field", "good-key")
+                        .addUserMetadata("field-valid", "true")
+                        .addUserMetadata("key-valid", "true")
+        ));
+
+        String badField = "prefix/bad-field";
+        String badFieldValue = "bad\u001dfield";
+        client.putObject(new PutObjectRequest(getTestBucket(), badField, new byte[0]).withObjectMetadata(
+                new S3ObjectMetadata().addEncodedUserMetadata("index-field", badFieldValue)
+                        .addUserMetadata("field-valid", "false")
+                        .addUserMetadata("key-valid", "true")
+        ));
+
+        try {
+            QueryObjectsRequest request = null;
+            QueryObjectsResult result = null;
+            // list the bad key with wrong prefix
+            /* Blocked by STORAGE-30527. Uncomment after fixed.*/
```

Comment:
Target fix version for STORAGE-30527 is 3.7.1.0. You could add version check so that the commented out code will be ignored before 3.7.1.0 is ready. Or just skip the test case before 3.7.1.0 is ready.

Reply:
Fix for STORAGE-30527 is almost ready and it is just decided to include the fix in 3.7.1.0. So added version check for 3.7.1.0.